includes/admin/class-wc-rest-payments-reader-controller.php

```diff
@@ -105,27 +105,24 @@ public function get_summary( $request ) {
 	 *
 	 * @param WP_REST_Request $request Full data about the request.
 	 * @return WP_HTTP_RESPONSE|WP_Error
+	 * @throws \RuntimeException Error collecting data.
 	 */
 	public function generate_print_receipt( $request ) {
 		try {
+			/* Collect the data, available on the server side. */
 			$payment_intent = $this->api_client->get_intent( $request->get_param( 'payment_id' ) );
-
 			if ( 'succeeded' !== $payment_intent->get_status() ) {
-				return rest_ensure_response( new WP_Error( 'generate_print_receipt_error', __( 'Invalid payment', 'woocommerce-payments' ), [ 'status' => 500 ] ) );
+				throw new \RuntimeException( __( 'Invalid payment', 'woocommerce-payments' ) );
```

Comment:
Shouldn't it be `Invalid payment intent`? Or is that the same thing?

Reply:
Good point!